src/www/ui/api/Controllers/UploadController.php

```diff
@@ -837,4 +837,42 @@ public function getGroupsWithPermissions($request, $response, $args)
     $res["permGroups"] = $finalPermGroups;
     return $response->withJson($res, 200);
   }
+
+  /***
+   * Remove the main license from the upload
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function removeMainLicense($request, $response, $args)
+  {
+    try {
+      $uploadId = intval($args['id']);
+      $shortName = $args['shortName'];
+      $returnVal = null;
+      $licenseDao = $this->container->get('dao.license');
+
+      $license = $licenseDao->getLicenseByShortName($shortName, $this->restHelper->getGroupId());
+
+      if (!$this->dbHelper->doesIdExist("upload", "upload_pk", $uploadId)) {
+        $returnVal = new Info(404, "Upload does not exist", InfoType::ERROR);
+      } else if ($license === null) {
+        $returnVal = new Info(404, "No license with shortname '$shortName' found.",
+          InfoType::ERROR);
+      }
```

Comment:
Please add a check with `ClearingDao::getMainLicenseIds()` to check if `$license` is actually the main license of the upload. If not, then return a 400

Reply:
Done